feelpp/feel/feelvf/matvec.hpp

```diff
@@ -820,6 +842,13 @@ class Vec
     //! expression is polynomial?
     bool isPolynomial() const { return fusion::accumulate( M_expr, true, GetIsPolynomial{} ); }
 
+    //! evaluate the expression without context
+    evaluate_type evaluate( bool, worldcomm_ptr_t const& ) const
+        {
+            evaluate_type res;
+            fusion::for_each( M_expr,vf::detail::evaluate_expression<evaluate_type>( res ) );
```

Comment:
how about using hana? that would simplify the code, make it more readable and avoid the class and it compiles faster according the hana's benchmarks